pkg/api/authn.go

```diff
@@ -86,22 +86,6 @@ func noPasswdAuth(realm string, config *config.Config) mux.MiddlewareFunc {
 				return
 			}
 
-			if config.HTTP.AllowReadAccess &&
```

Comment:
This check is saying if read-only access is enabled and mutual tls is enabled and methods are write, deny them. Is this getting subsumed by access control policy now?

Reply:
handling mutual auth is not in this PR, will arrive in the Nicol's next one, maybe we should merge them because of this?

you are right, this is a bug.

Reply:
besides tls mutual auth, the rest is handled in the authz.